### ISSUE TITLE
Support for Axios 1.3.0

### ISSUE THREAD

**Issue by Supernova989** [nestjs/axios#663]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

The issue appears when the axios 1.2.3+ is explicitly installed in the project. Then, code like this isn't working:

```typescript
import { HttpService } from '@nestjs/axios';
import { Injectable } from '@nestjs/common';
import { AxiosInstance } from 'axios';

@Injectable()
export class MyService {
  private http: AxiosInstance;

  constructor(private readonly httpService: HttpService) {
    this.http = httpService.axiosRef;
  }
}
```

The reason is that typings and structure for AxiosInstance starting from 1.2.3 and higher is different from what goes with `@nestjs/axios` right now.

### Describe the solution you'd like

The internal dependency for Axios is 1.3.0

### What is the motivation / use case for changing the behavior?

Typically you grab instances from Axios directly - not from sub-dependences of install packages (`@nestjs/axios` in this case).

**Comment by micalevisk** [comment]
this is be solved (sorta) by PR #659 

but yeah, if we could make it work with `axios@^1` it would be better for DX

**Comment by kamilmysliwiec** [comment]
I'm going to move axios to peerDependencies
